src/viking.js

```diff
@@ -1,11 +1,138 @@
 // Soldier
-class Soldier {}
+class Soldier {
+    constructor(health, strength) {
+        this.health = health;
+        this.strength = strength;
+    }
 
+    attack() {
+        return this.strength;
+    }
+
+    receiveDamage(damage) {
+        this.health = this.health - damage;
+    }
+
+}
 // Viking
-class Viking {}
+class Viking extends Soldier {
+    constructor(name, health, strength){
+        super(name, health, strength);
+        this.name= name;
+        this.health = health;
+        this.strength = strength;
+    }
+
+    receiveDamage(damage) {
+        this.health = this.health - damage;
+            if (this.health <= 0) {
+                return `${this.name} has died in act of combat`; 
+            } else {
+        return `${this.name} has received ${damage} points of damage`;
+        }
+    }
+
```

Comment:
Nice work Yann, I just want to point out that line 28-32 can be written using the ternary operator
```
return this.health <= 0 ? return `${this.name} has died in act of combat`: return `${this.name} has received ${damage} points of damage`;